proptest/src/sugar.rs

```diff
@@ -734,18 +728,15 @@ macro_rules! prop_compose {
 /// ```
 #[macro_export]
 macro_rules! prop_assert {
-    ($cond:expr) => {
+    ($cond:expr) => { {
         $crate::prop_assert!($cond, concat!("assertion failed: ", stringify!($cond)))
-    };
+    } };
 
-    ($cond:expr, $($fmt:tt)*) => {
-        if !$cond {
-            let message = format!($($fmt)*);
-            let message = format!("{} at {}:{}", message, file!(), line!());
-            return ::core::result::Result::Err(
-                $crate::test_runner::TestCaseError::fail(message));
-        }
-    };
+    ($cond:expr, $($fmt:tt)*) => { {
+        let message = format!($($fmt)*);
+        let message = format!("{} at {}:{}", message, file!(), line!());
+        assert!($cond, "{}", message);
```

Comment:
Should we just put the actual message string here?

Reply:
Going with the actual message. std::fmt is not linked.